apps/dashboard/app/lib/smart_attributes/attributes/auto_environment_variable.rb

```diff
@@ -0,0 +1,37 @@
+module SmartAttributes
+    class AttributeFactory
+      extend AccountCache
+  
+      def self.build_auto_environment_variable(opts = {})
+        Attributes::AutoEnvironmentVariable.new('auto_environment_variable', opts)
+      end
+    end
+  
+    module Attributes
+      class AutoEnvironmentVariable < Attribute
+        def initialize(id, opts = {})
+          super
+  
+          @key = @opts.delete(:key)
+          # reset the id to be unique from other auto_environment_variable_*
+          @id = @key ? "#{id}_#{normalize_key(@key)}" : id
+        end
+
+        def widget
+          'text_field'
+        end
+        
+        def label(*)
+          (@key|| 'Environment Variable').to_s
```

Comment:
I think the way you had this initially was right, though maybe it should be:

```ruby
 (opts[:label] || "Environment Variable: #{@key.to_s}").to_s
```

People love to relabel stuff, so we should allow for that (recall the path_selector bugs in 3.0 ppl can't relabel or require it).